app/components/wallet-overview/wallet-overview-redesigned.tsx

```diff
@@ -0,0 +1,214 @@
+
+import { gql } from "@apollo/client"
+import {
+  useHideBalanceQuery,
+  useWalletOverviewScreenQuery,
+  WalletCurrency,
+} from "@app/graphql/generated"
+import { useIsAuthed } from "@app/graphql/is-authed-context"
+import { useDisplayCurrency } from "@app/hooks/use-display-currency"
+import { toBtcMoneyAmount, toUsdMoneyAmount } from "@app/types/amounts"
+// import { testProps } from "@app/utils/testProps"
+import { makeStyles, Text, useTheme } from "@rneui/themed"
+
+import { GaloyCurrencyBubble } from "../atomic/galoy-currency-bubble"
+import { GaloyIconButton } from "../atomic/galoy-icon-button"
+import { GaloyIcon } from "../atomic/galoy-icon"
+
+type HidableAreaProps = {
+  hidden: boolean
+  style?: StyleProp<ViewStyle>
+  children: React.ReactNode
+}
+
+const Loader = ({ styles }: { styles: StyleProp<ViewStyle> }) => (
+  <View style={styles}>
+    <ContentLoader
+      height={"20%"}
+      width={"80%"}
+      speed={1.2}
+      backgroundColor="#f3f3f3"
+      foregroundColor="#ecebeb"
+    >
+      <Rect x="0" y="0" rx="4" ry="4" width="100%" height="100%" />
+    </ContentLoader>
+  </View>
+)
+
+const HidableArea = ({ hidden, style, children }: HidableAreaProps) => {
+  return (
+    <TouchableHighlight style={style} underlayColor="#ffffff00">
+      <>{hidden ? <Text>****</Text> : children}</>
+    </TouchableHighlight>
+  )
+}
+
+gql`
+  query walletOverviewScreen {
+    me {
+      defaultAccount {
+        id
+        btcWallet @client {
+          id
+          balance
+        }
+        usdWallet @client {
+          id
+          balance
+        }
+      }
+    }
+  }
+`
+
+type Props = {
+  loading: boolean
+  setIsStablesatModalVisible: (value: boolean) => void
+}
+
+const WalletOverviewRedesigned: React.FC<Props> = ({
+  loading,
+  setIsStablesatModalVisible,
+}) => {
+  const isAuthed = useIsAuthed()
+  const { theme } = useTheme()
+  const styles = useStyles(theme)
+  const { data: dataHideBalance } = useHideBalanceQuery()
+  const hideBalance = dataHideBalance?.hideBalance || false
+  const [visible, setVisible] = useState(hideBalance)
+
+  const { data } = useWalletOverviewScreenQuery({ skip: !isAuthed })
+  const { formatMoneyAmount, displayCurrency, moneyAmountToDisplayCurrencyString } =
+    useDisplayCurrency()
+
+  let btcInDisplayCurrencyFormatted = "$0.00"
+  let usdInDisplayCurrencyFormatted = "$0.00"
+  let btcInUnderlyingCurrency = "0 sat"
+  let usdInUnderlyingCurrency: string | undefined = undefined
+
+  useEffect(() => {
+    setVisible(hideBalance)
+  }, [hideBalance])
+
+  if (isAuthed) {
+    const btcWalletBalance = toBtcMoneyAmount(
+      data?.me?.defaultAccount?.btcWallet?.balance ?? NaN,
+    )
+
+    const usdWalletBalance = toUsdMoneyAmount(
+      data?.me?.defaultAccount?.usdWallet?.balance ?? NaN,
+    )
+
+    btcInDisplayCurrencyFormatted =
+      moneyAmountToDisplayCurrencyString(btcWalletBalance) ?? "..."
```

Comment:
Does `moneyAmountToDisplayCurrencyString` ever return a falsy value? Otherwise I don't think we will get the "..." ever.